File: smoothie-sample/src/main/java/com/example/smoothie/SimpleActivity.java
package com.example.smoothie;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import com.example.smoothie.deps.ContextNamer;
import javax.inject.Inject;
import toothpick.Scope;
import toothpick.Toothpick;
import toothpick.smoothie.module.SmoothieActivityModule;

public class SimpleActivity extends Activity {

  private Scope scope;

  @Inject ContextNamer contextNamer;
  @BindView(R.id.title) TextView title;
  @BindView(R.id.subtitle) TextView subTitle;
  @BindView(R.id.hello) Button button;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    scope = Toothpick.openScopes(getApplication(), this);
    scope.installModules(new SmoothieActivityModule(this));
    super.onCreate(savedInstanceState);
    Toothpick.inject(this, scope);
    setContentView(R.layout.simple_activity);
    ButterKnife.bind(this);
    title.setText(contextNamer.getApplicationName());
    subTitle.setText(contextNamer.getActivityName());
    button.setText("click me !");
  }

  @OnClick(R.id.hello)
  @SuppressWarnings("unused")
  void startNewActivity() {
    startActivity(new Intent(this, RxMVPActivity.class));
  }

  @Override
  protected void onDestroy() {
    Toothpick.closeScope(this);
    super.onDestroy();
  }
}

File: smoothie-sample/src/main/java/com/example/smoothie/kotlin/SimpleActivity.kt
package com.example.smoothie.kotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.fragment.app.FragmentActivity
import com.example.smoothie.R
import com.example.smoothie.kotlin.deps.ContextNamer
import toothpick.Toothpick
import toothpick.kotlin.androidx.inject
import toothpick.smoothie.module.SmoothieAndroidXActivityModule

class SimpleActivity : FragmentActivity() {
    // Will use activity instance as scope
    private val contextNamer: ContextNamer by inject()

    private val title by lazy { findViewById<TextView>(R.id.title) }
    private val subTitle by lazy { findViewById<TextView>(R.id.subtitle) }
    private val button by lazy { findViewById<Button>(R.id.hello) }

    override fun onCreate(savedInstanceState: Bundle?) {
        val scope = Toothpick.openScopes(application, this)
        scope.installModules(SmoothieAndroidXActivityModule(this))
        super.onCreate(savedInstanceState)
        setContentView(R.layout.simple_activity)

        title.text = contextNamer.applicationName
        subTitle.text = contextNamer.activityName
        button.apply {
            text = "click me !"
            setOnClickListener { startActivity(Intent(this@SimpleActivity, RxMVPActivity::class.java)) }
        }
    }

    override fun onDestroy() {
        Toothpick.closeScope(this)
        super.onDestroy()
    }
}
